src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[10];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity(size + 1);
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        ensureCapacity(size + 1);
+        try {
+            System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        } catch (IndexOutOfBoundsException e) {
+            throw new ArrayListIndexOutOfBoundsException("Can't add element to this index");
+        }
```

Comment:
instead of try catch better check index before you try to execute arraycopy, the same for other similar cases